File: models/frame.rb
class Frame
  def initialize(roll_one, roll_two, roll_three)
    @frame_points = []
    add_points(roll_one, roll_two, roll_three)
  end

  def add_points(roll_one, roll_two, roll_three)
    @frame_points << roll_one
    @frame_points << roll_two
    @frame_points << roll_three
  end

  def frame_points
    return @frame_points
  end

  def strike?
    @frame_points[0] == 10
  end

  def spare?
    @frame_points[0] != 10 && @frame_points.sum == 10
  end
end

File: models/frame.js
class Frame {
  constructor() {
    this.points = [];
  }

  addPoints(rollOne, rollTwo = 0, rollThree = 0) {
    this.points.push(rollOne);
    this.points.push(rollTwo);
    this.points.push(rollThree);
  }

  framePoints() {
    return this.points;
  }

  strike() {
    return this.points[0] === 10
  }

  spare() {
    return this.points[0] !== 10 && (this.points[0] + this.points[1]) === 10;
  }
}

module.exports = Frame;
